pennylane/plugins/default_qubit.py

```diff
@@ -412,16 +414,24 @@ def sample(self, observable, wires, par):
     def _get_operator_matrix(self, operation, par):
         """Get the operator matrix for a given operation or observable.
 
+        If the inverse was defined for an operation or observable that requires parameters, returns the
+        operator matrix obtained using negated paramaters.
+
         Args:
           operation    (str): name of the operation/observable
           par (tuple[float]): parameter values
         Returns:
           array: matrix representation.
         """
-        A = {**self._operation_map, **self._observable_map}[operation]
-        if not callable(A):
-            return A
-        return A(*par)
+
+        operation_map = {**self._operation_map, **self._observable_map}
+
+        if operation.endswith(Operation.string_for_inverse):
```

Comment:
I see that `"PauliX.inv"` is the natural generalization of our current plugin interface. But wouldn't it probably be better to have the inversion as a separate boolean parameter to avoid having to look at the strings? What do you think, @josh146 @co9olguy ?

Reply:
There was a lot of discussion behind this :laughing: choosing to serialize inversion into the name was done for two reasons:

1. A desire not to break the plugin API if possible

2. The fact that most hardware compatible frameworks don't have a general concept for inversion, but instead do it on a gate by gate basis. For example, PyQuil and Qiskit both manually define `Sdg` and `Tdg`, but have no concept of the inverse of a CNOT, for example.

  By doing it this way, it scales a bit more naturally. A plugin can declare it supports `S.inv`, without requiring it to support inverting all gates.

Perhaps we could combine the two approaches - have it in the capabilities dictionary for all gates, or in the name only for specific gates.

Reply:
Okay, that makes sense - then we should stick with the logic that is implemented now! Combining the approaches will be super-confusing I fear.

But I would still have the implementation on the `Operation` side as I outlined above, because appending and removing `.inv` from the operation name does not feel like good style.

Reply:
btw, I had no idea there was a Python `str.endswith()` method :boom: